File: my-folder/problems/3sum/solution.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        int n = nums.size();
        vector<vector<int>> v;
        set<vector<int>> s;
        for(int i = 0; i < n; i++)
        {
            int l = i+1;
            int r = n-1;
            int x = nums[i];
            while(l<r)
            {
                if(x+nums[l]+nums[r] == 0)
                {
                    vector<int> temp = {x,nums[l],nums[r]};
                    s.insert(temp);
                    l++;
                    r--;
                }
                else if(x+nums[l]+nums[r] < 0)
                {
                    l++;
                }
                else
                    r--;
            }
        }
        for(auto x: s)
        {
            v.push_back(x);
        }
        return v;
    }
};

File: my-folder/problems/3sum/solution.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        List <List<Integer>>  arr = new ArrayList<List<Integer>>();
        Set <List<Integer>> s = new HashSet<List<Integer>>();
        Arrays.sort(nums);
        for(int i=0; i< nums.length-2; i++)
        {  
            int l= i+1; 
            int r= nums.length-1;
            while(l < r)
            {
                if((nums[i] + nums[l] + nums[r]) == 0)
                {
                    List<Integer> li = new ArrayList<Integer>(Arrays.asList(nums[i], nums[l], nums[r]));
                    s.add(li);
                    l++;
                    r--;
                }
                else if((nums[i] + nums[l] + nums[r]) < 0)
                    l++;
                else
                    r--;
            }
        }
        for(List<Integer> it : s)
        {
            arr.add(it);
        } 
        return arr;                  
    }
}